gateway/ln-gateway/src/lib.rs

```diff
@@ -1108,48 +1113,51 @@ impl Gateway {
             }
         };
 
+        let per_federation_routing_fees: HashMap<FederationId, RoutingFees> =
+            per_federation_routing_fees
+                .unwrap_or_default()
```

Comment:
I don't think we want to use `unwrap_or_default` here. Ideally we should only update the federation's routing fees is routing fees were supplied in `SetPayloadConfiguration` (i.e `per_federation_routing_fees.is_some()`. If I'm reading this correctly, if a gateway operator only wants to update their route hints or another unrelated value, the fees are going to be overwritten with whatever the default routing fees are.